### DISCUSSION TITLE
Is directory flattening broken?

### DISCUSSION THREAD

**Discussion by gerroon** [bpatrik/pigallery2#852] [Q&A]
Hi

I have folders with folders with thousands of files in them. When I do ungroup directory flattening, I small number of files shown as result. Say 80 files out of 20k or sometimes flattening shows none. It seems to me that it is not recursing deep into folders.

Now I set my max media search result to 4000, so it cant be that. This used to work but I do not remember at what point this has changed.

I use docker :latest on Linux. 

So here is a folder without flattening, this folder probably has over 1000 files

![image](https://github.com/bpatrik/pigallery2/assets/8519469/d03f44f0-b3a5-44be-82e6-66076bae7afe)

When I flatten I only get 175. Is it possible that it is listing based on some percentage instead of the actual limit number?
![image](https://github.com/bpatrik/pigallery2/assets/8519469/a9b7b929-0bda-439c-912d-e9eec17adc49)

**Comment by bpatrik** [comment]
Moved this to a bug #853 .

I will take a look at some point

**Comment by bpatrik** [comment]
I can't reproduce it on the latest edge build.

**Reply by bpatrik** [reply]
My random  guess to this problem is that your directory is not indexed. Flattening only works on indexed directory.
